src/DocumentationViewer.js

```diff
@@ -0,0 +1,219 @@
+import GObject from "gi://GObject"
+import Gtk from "gi://Gtk";
+import Gio from "gi://Gio"
+import GLib from "gi://GLib";
+import Adw from "gi://Adw";
+import WebKit from "gi://WebKit"
+
+import resource from "./DocumentationViewer.blp";
+
+Gio._promisify(
+  WebKit.WebView.prototype,
+  "evaluate_javascript",
+  "evaluate_javascript_finish"
+);
+
+Gio._promisify(
+  Gio.File.prototype,
+  "enumerate_children_async",
+  "enumerate_children_finish",
+);
+
+Gio._promisify(
+  Gio.FileEnumerator.prototype,
+  "next_files_async",
+  "next_files_finish",
+);
+
+Gio._promisify(
+  Gio.File.prototype,
+  "load_contents_async",
+  "load_contents_finish",
+);
+
+export default function DocumentationViewer({
+  window:application_window,
+  application
+  }) {
+  const builder = Gtk.Builder.new_from_resource(resource);
+  const window = builder.get_object("documentation_viewer");
+
+  const container = builder.get_object("webview_container");
+  const webview = new WebKit.WebView();
+  const listbox = builder.get_object("listbox");
+
+  const sidebar_button = builder.get_object("sidebar_button");
+  const search_button = builder.get_object("search_button");
+  const search_entry = builder.get_object("search_entry");
+  const search_bar = builder.get_object("search_bar");
+  const back = builder.get_object("back");
+  const forward = builder.get_object("forward");
+
+  const base_path = "/app/share/doc";
+
+  webview.load_uri(`file://${base_path}/gtk4/index.html`);
```

Comment:
in general, never use string concat for building paths, or other well known formats
there are many pitfalls

Please use https://gjs-docs.gnome.org/glib20~2.0/glib.uri#function-build

Reply:
Could you give an example of how I should build the uri, when I'm using ``GLib.Uri.build``, the actual path will still have to made using string concats right? Something like this
```js
const base_path = "/app/share/doc";
const path = `${base_path}/gtk4/index.html`;
const uri = GLib.Uri.build(
  GLib.UriFlags.NONE,
  "file",
  null,
  null,
  -1,
  path,
  null,
  null,
);
```
(Also the code above returns `file:/app/share/doc/gtk4/index.html` as supposed to `file:///app/share/doc/gtk4/index.html`, I'm not sure if I'm using it correctly)

Reply:
hmm my bad, this is the way to go:

`new Gio.File('/app/share/doc').resolve_relative_path('gtk4/index.html').get_uri()`